src/AppInstallerCLICore/Workflows/ArchiveFlow.cpp

```diff
@@ -4,8 +4,18 @@
 #include "ArchiveFlow.h"
 #include "winget/Archive.h"
 
+using namespace AppInstaller::Manifest;
+
 namespace AppInstaller::CLI::Workflow
 {
+    namespace
+    {
+        bool RelativePathContainsEscapeString(const std::string& relativePath)
+        {
+            return relativePath.find("..") != std::string::npos;
```

Comment:
Is there no API for this kind of check? It would be good to not have to think of every possible case ourselves.

Maybe instead of checking that the relative path doesn't contain anything like this, you could check that the final path looks like you expect. Like `(basePath + relativePath).resolve().starts_with(basePath)` (made up syntax).

Reply:
I didn't find any API that could check for this specifically. From what I was able to find, it seems like ".." was the only string that could be used to access parent directories.  

Regarding the check, the reason why I chose to validate just the RelativeFilePath is because this is the only input the user could have to affect the installerPath since the basePath is created by us. This also allows us to run checks without having to start the download/install flow. However, if you feel strongly about this, I could also add this check to the fullPath once available in the install flow to be extra safe.

Reply:
[std::filesystem:canonical](https://en.cppreference.com/w/cpp/filesystem/canonical) will help resolve . and .. and even symbolic link for us.
It may be also helpful to move the related "check if a relative path is still under expected folder" code to filesystem.cpp under CommonCore for future "launch" related work(which also needs this check I think)

Reply:
I also feel like a std::find may cause a very odd edge case where a publisher or an error in their build pipeline may create a filename like `File.Major.Minor..Build`

It may be a good use case not only for canonical as Yao suggested, but perhaps even [std::filesystem:relative](https://en.cppreference.com/w/cpp/filesystem/relative)

Reply:
Created a method called `PathEscapesBaseDirectory` in `Filesystem.cpp` that checks if the canonical form of a path fails to contain the base path. If true, then we know that the path points to a location outside of the provided base directory